Found with  	OpenDJ 7.0.0-SNAPSHOT (35d9249c9cf)

This issue seems to happen a bit randomly, whatever the upgrade path is.

We set up 2 servers, one with data.
We configure replication between them and initialize it.
Then we set confidentiality on for changelogs on both servers:

/DJ_ENCRYPT1/opendj/bin/dsconfig -h openam.example.com -p 4451 -D "cn=myself" -w password -X set-replication-server-prop --provider-name "Multimaster Synchronization" --set confidentiality-enabled:true -n

DJ_ENCRYPT2/opendj/bin/dsconfig -h openam.example.com -p 4452 -D "cn=myself" -w password -X set-replication-server-prop --provider-name "Multimaster Synchronization" --set confidentiality-enabled:true -n



We check replication is working by doing LDAP operations on both servers.
Then we stop and upgrade one server.
We start it again and add some entries to check replication is working fine.
At that time, we compare changelog content and sometimes they are not in sync (even after more than 5minutes wait.)


Content of servers differs, differences can be found at: /root/workspace/OpenDJ-7.0.x/tests_ds_upgrade_paths_je_part1/results/20190926-150751/replication_group3/Upgrade/DJ_ENCRYPT1/opendj/tmp/diff_Encrypted_Replication_Topology_changelog_DJ_ENCRYPT1_DJ_ENCRYPT2.ldif 	
17:24:39.319 	INFO 	dn: changeNumber=10,cn=changelog
changetype: delete

dn: changeNumber=11,cn=changelog
changetype: delete

dn: changeNumber=12,cn=changelog
changetype: delete

dn: changeNumber=13,cn=changelog
changetype: delete

dn: changeNumber=14,cn=changelog
changetype: delete

dn: changeNumber=15,cn=changelog
changetype: delete

dn: changeNumber=16,cn=changelog
changetype: delete

dn: changeNumber=9,cn=changelog
changetype: delete



I have run this test on my laptop and most of the time, it succeeds while it fails everytime on jenkins.

Test is:

./run-pybot.py -s replication_group3.Upgrade -t encrypted_replication_topology opendj



it seems that what is reproducible is the entries that are missing.